Realization: hashCode, equals, clone, toString, Factory, BooleanFieldFormat, hidden, defaultValue

Andrew7882 opened the pull request.

Andrew7882 committed:
Realization of equals(), hashCode(), clone()

Andrew7882 committed:
Realization of Tests for equals(), hashCode(), clone()

Andrew7882 committed:
Realization of Tests for equals(), hashCode(), clone()
Realization of Abstract Factory pattern:
- abstract base factory class FieldFormatCreator;
- concrete creators StringFieldFormatCreator and IntFieldFormatCreator.

Andrew7882 committed:
Realization of Tests for equals(), hashCode(), clone()
Realization of Abstract Factory pattern:
- abstract base factory class FieldFormatCreator;
- concrete creators StringFieldFormatCreator and IntFieldFormatCreator.
BooleanFieldFormat was added.

Andrew7882 committed:
Realization of Tests for equals(), hashCode(), clone()
Realization of Abstract Factory pattern:
- abstract base factory class FieldFormatCreator;
- concrete creators StringFieldFormatCreator and IntFieldFormatCreator.
BooleanFieldFormat was added.
Fields 'hidden' and 'defaultValue' was added.

Andrew7882 committed:
1.
Realization of Tests for equals(), hashCode(), clone()
Realization of Abstract Factory pattern:
- abstract base factory class FieldFormatCreator;
- concrete creators StringFieldFormatCreator and IntFieldFormatCreator.
BooleanFieldFormat was added.
Fields 'hidden' and 'defaultValue' was added.
Validtaros still in progress.
2.
equals, hashCode, toString, clone was added to TableFormat.
Field for minimal and maximal rows quantity was added.

Andrew7882 committed:
1.
Realization of Tests for equals(), hashCode(), clone()
Realization of Abstract Factory pattern:
- abstract base factory class FieldFormatCreator;
- concrete creators StringFieldFormatCreator and IntFieldFormatCreator.
BooleanFieldFormat was added.
Fields 'hidden' and 'defaultValue' was added.
Validtaros still in progress.
2.
equals, hashCode, toString, clone was added to TableFormat.
Field for minimal and maximal rows quantity was added.

AndreyPlis commented on src/main/java/com/tibbo/datatable/FieldFormatFactory.java:
```diff
@@ -1,23 +1,30 @@
+        FieldFormat ff = null;
+        /*if (type == FieldFormat.STRING_FIELD)
             ff = new StringFieldFormat();
         else if (type == FieldFormat.INTEGER_FIELD)
```
закоментированный код надо удалять

AndreyPlis commented on src/main/java/com/tibbo/datatable/FieldFormatCreator.java:
```diff
@@ -4,14 +4,12 @@
-                }
-            }
+        switch (attr.length){
+            case 5: ff.setDefaultValue(ff.valueFromString(attr[4]));
```
почему 5,4,3,2,1? а если я передам в другом порядке параметры?

Andrew7882 committed:
1.
Realization of Tests for equals(), hashCode(), clone()
Realization of Abstract Factory pattern:
- abstract base factory class FieldFormatCreator;
- concrete creators StringFieldFormatCreator and IntFieldFormatCreator.
BooleanFieldFormat was added.
Fields 'hidden' and 'defaultValue' was added.
Validtaros still in progress.
2.
equals, hashCode, toString, clone was added to TableFormat.
Field for minimal and maximal rows quantity was added.
FieldValidator class was added with NullFieldValidator, LimitsFieldValidator, RegexFieldValidator children.

AndreyPlis commented on src/main/java/com/tibbo/datatable/BooleanFieldFormatCreator.java:
```diff
@@ -0,0 +1,10 @@
+
+public class BooleanFieldFormatCreator extends FieldFormatCreator{
+    @Override
+    public FieldFormat create(String... attr){
```
здесь лучше не использовать vararg

AndreyPlis commented on src/main/java/com/tibbo/datatable/FieldFormat.java:
```diff
@@ -40,18 +43,70 @@ public void setNullable(Boolean nullable) {
+            return false;
+        }
+        FieldFormat<T> fieldFormat = (FieldFormat<T>)o;
+        return (name == fieldFormat.name || (name != null && name.equals(fieldFormat.getName())) )
```
лучше эту конструкцию разнести на отдельные if

AndreyPlis commented on src/main/java/com/tibbo/datatable/FieldFormat.java:
```diff
@@ -40,18 +43,70 @@ public void setNullable(Boolean nullable) {
     @Override
     public String toString() {
-        throw new UnsupportedOperationException();
+        //throw new UnsupportedOperationException();
```
удаляй закомментированные строки
